clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java

```diff
@@ -371,6 +375,38 @@ public synchronized AbstractStorageEngine openStore(
     return engine;
   }
 
+  public synchronized void checkWhetherStoragePartitionsShouldBeKeptOrNot(SafeHelixManager manager) {
+    if (getStorageEngineRepository() == null || manager == null) {
+      return;
+    }
+    for (AbstractStorageEngine storageEngine: getStorageEngineRepository().getAllLocalStorageEngines()) {
+      String storageEngineName = storageEngine.getStoreVersionName();
+      String storeName = Version.parseStoreFromKafkaTopicName(storageEngineName);
+      PropertyKey.Builder propertyKeyBuilder =
+          new PropertyKey.Builder(configLoader.getVeniceClusterConfig().getClusterName());
+      SafeHelixDataAccessor helixDataAccessor = manager.getHelixDataAccessor();
+      IdealState idealState = helixDataAccessor.getProperty(propertyKeyBuilder.idealStates(storeName));
+
+      if (idealState == null) {
+        return;
+      }
+
+      Set<Integer> idealStatePartitionIds = new HashSet<>();
+
+      idealState.getPartitionSet().stream().forEach(partitionId -> {
+        idealStatePartitionIds.add(Integer.parseInt(partitionId));
+      });
+      Set<Integer> storageEnginePartitionIds = storageEngine.getPartitionIds();
+
+      for (Integer storageEnginePartitionId: storageEnginePartitionIds) {
+        if (idealStatePartitionIds.contains(storageEnginePartitionId)) {
```

Comment:
This also isn't quite right.  The ideal state is the superset of all partitions for the store.  All partitions ID's in the storageEngine will be in the superset.

We need to parse more information from the ideal state.  Specifically, for each partition that is on this node, we need to determine if the ideal state for that partition id contains the node name.

To better paint the picture, an ideal state is a big json document that looks like this:

>    "Test_Store_Migration_Demo_v1_0": {
>       "lor1-app56585.prod.linkedin.com_1690": "LEADER",
>       "lor1-app56614.prod.linkedin.com_1690": "STANDBY",
>       "lor1-app110448.prod.linkedin.com_1690": "STANDBY"
>     },
>     "Test_Store_Migration_Demo_v1_1": {
>       "lor1-app56586.prod.linkedin.com_1690": "LEADER",
>       "lor1-app71895.prod.linkedin.com_1690": "STANDBY",
>       "lor1-app111181.prod.linkedin.com_1690": "STANDBY"
>     },

Here, for partitions 1 and 0, we have three hosts assigned roles for that partition.  What we should do is, does the current host we're working with reside in these lists?

Reply:
I have written the comparison and retrieved the current host from `manager.getInstanceName().`

Reply:
The code logic here to consult the ideal state's information of partitions and the hosts they are assigned to [in the list discussed] has been added, and comparison to current host to determine how to keep the partition has been integrated.